depends/toolchain.cmake.in

```diff
@@ -84,15 +84,19 @@ set(OTOOL "@OTOOL@")
 
 set(CMAKE_FIND_ROOT_PATH "@depends_prefix@")
 set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
-set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY BOTH)
```

Comment:
I've added an explanation comment: https://github.com/hebasto/bitcoin/blob/b0a1bae9f81ff59ca63915b293e9202a9609a4e8/depends/toolchain.cmake.in#L95-L99

Reply:
De-globalized. An explanation comment added: https://github.com/hebasto/bitcoin/blob/c6c2cf964abf01b0cb21e4c6fcf90fd18ca72b2e/src/qt/CMakeLists.txt#L18-L26

Reply:
I agree with this change, moving it to CMakeLists.txt to de-globalized, but it was copied and left in `toolchain.cmake.in`, is this correct for when the dependencies are built?

Also, `set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY BOTH)` is required not only for cross-compilation but also for Mac or Apple devices, as I found on "_Professional CMake - A Practical Guide by Craig Scott_" (2023):

_On Apple platforms, consider carefully before setting `CMAKE_FIND_ROOT_PATH_MODE_LIBRARY` to `ONLY`, as libraries may be built as fat binaries which support multiple target platforms. These fat binaries may not reside under target platform-specific paths, so it may still be necessary to search host platform paths to find them._

Do we need `set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)` to change it to `BOTH` as well?

Reply:
> Do we need `set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)` to change it to `BOTH` as well?

I don't think so, because our both native macOS CI jobs are green. And we set the `CMAKE_FIND_ROOT_PATH` only for cross-compiling (please note that the `CMAKE_FIND_ROOT_PATH_MODE_*` variables values affect how the `CMAKE_FIND_ROOT_PATH` is applied during searching routines).

Reply:
I spent most of yesterday poking at this. It's extremely annoying. I don't understand why the platform libs are lumped in here.

Is QT the only offender? Do other cmake projects also search for their platform dependencies like this? Afaik it makes `CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY` completely useless.

If qt is the only lib with this behavior, I'm wondering if we should patch around it and/or upstream a change?

Building with `-DQt5_EXCLUDE_STATIC_DEPENDENCIES:BOOL=ON` _almost_ fixes the problem by letting us link the required libs ourselves. Except that the static plugins don't honor it :\

Reply:
> Is QT the only offender?

It seems so.

Reply:
> If qt is the only lib with this behavior, I'm wondering if we should patch around it and/or upstream a change?

The Qt active development happens for version 6 now. Suggesting to postpone this decision until we also bump our Qt package to the next major version. I'm assuming that things might change to better in there.

Reply:
Heh, we've been burned too many times by qt in the past to assume that it will actually become more reasonable in the future. It's more likely that they'll invent yet another wonky buildsystem :p

As far as this goes though, does the fact that this lives in `src/qt/CMakeLists.txt` mean that the `CMAKE_FIND_ROOT_PATH_MODE_LIBRARY` gets scoped to QT only? Or does that apply the option globally?

Reply:
> does the fact that this lives in `src/qt/CMakeLists.txt` mean that the `CMAKE_FIND_ROOT_PATH_MODE_LIBRARY` gets scoped to QT only?

Yes, it does.

Reply:
Ok, great. In that case I care much less about this and am ok with punting.